File: python/balanced_parentheses_spec.py
from balanced_parentheses import balanced_parentheses

# print(balanced_parentheses("()")) == True
# print(balanced_parentheses("())")) == False
# print(balanced_parentheses("(){}")) == True
# print(balanced_parentheses("({})")) == True
# print(balanced_parentheses("(){}[])")) == False
print(balanced_parentheses("][")) == False
# print(balanced_parentheses("(({}")) == False
# print(balanced_parentheses("}([]){")) == False
print(balanced_parentheses("{[()]({})}()")) == True

File: js/balancedParenthesesSpec.js
const { balancedParentheses } = require("./balancedParentheses")

console.log(balancedParentheses("()")) // should return true
console.log(balancedParentheses("())")) // should return false
console.log(balancedParentheses("(){}")) // should return true
console.log(balancedParentheses("({})")) // should return true
console.log(balancedParentheses("(){}[])")) // should return false
console.log(balancedParentheses("][")) // should return false
console.log(balancedParentheses("(({}")) // should return false
console.log(balancedParentheses("}([]){")) // should return false
console.log(balancedParentheses("]]")) // should return false
console.log(balancedParentheses("{[()")) // should return false
console.log(balancedParentheses("{[()]({})}()")) // should return true
